Kernel/Arch/init.cpp

```diff
@@ -308,6 +290,12 @@ extern "C" [[noreturn]] UNMAP_AFTER_INIT void init([[maybe_unused]] BootInfo con
     InterruptManagement::initialize();
     ACPI::initialize();
 
+#if ARCH(RISCV64)
+    // FIXME: Unflatten the device tree and use it for device discovery
+    if (kernel_command_line().contains("dump_fdt"sv))
```

Comment:
One small nitpick:

afaict `CommandLine::contains` is never used directly anywhere else.
So you should probably add a wrapper in CommandLine.cpp just like for other command line args.

Reply:
I mentioned this to Andrew a while back and he was of the opinion that it might be fine and that the command line interface seems to be misnomer-ed and overgrown at some points, so "Won't Fix" I guess

Reply:
OK, then keep it like that.

Reply:
Yeah we should probably refactor the command line class to be more reasonable at some point :yak shrug: